spec/controllers/books_controller_spec.rb

```diff
@@ -0,0 +1,30 @@
+describe BooksController do
+  describe 'GET #index' do
+    subject(:make_request) { get :index, params: params }
+
+    let(:params) { { topic_id: topic_id } }
+    let(:topic_id) { 1 }
+
+    context 'when a valid topic is given' do
```

Comment:
You repetead this context and the other in `spec/controllers/topics_controller_spec.rb`. You can move all this code to a shared_example for this.

Reply:
Isn't it different on the swagger exporter ?
I believe they then have some method calls on the books controller, as a way to test the integration for RSpec.